### ISSUE TITLE
Unique custom domains in an ingress do not get a unique GLBC host

### ISSUE THREAD

**Issue by maleck13** [kcp-dev/kcp-glbc#492]
Priority: Medium

## Expected Behavior
* If I have an ingress with two custom hosts (IE apps.example.com and apps.whatever.com) I should get two glbc generated hosts instead of the same one being used for both rules

## Actual Behavior
* I get the same host for both rules

### Steps to Reproduce the Problem

1.

## Suggested Resolution
* A work around here is to create two ingresses 
* We should support more than 1 host being specified in an ingress as ingresses also support this behaviour

**Comment by maleck13** [comment]
@philbrookes I guess ultimately it would make sense for GLBC to be able to set multiple managed hosts per ingress, thoughts?

**Comment by philbrookes** [comment]
we did originally do this intentionally. Is the reason to have multiple per ingress to avoid the chance of colliding paths on the same host again? (the same reason we can't use a namespace scoped generated host).

Everything in the ingress will be scheduled to the same sync targets, so there is no DNS resolution reason for unique generated hosts per rule, right?

**Comment by maleck13** [comment]
Yes exactly if we use the same GLBC host in an ingress with mutliple hosts there could be a clash at the path level
